### PR TITLE
Add wizard loadouts, misc wiz features/fixes

### LINKED ISSUE #12668: Plasmaman Wizards die at round start
Plasmaman Wizards don't spawn with any sort of envirosuit or internals. They promptly die.

**What did you expect to happen**: 
A Plasmaman wizard to spawn with special envirosuit and internal tank.

**What happened instead**:
The fastest wizard round in the west.

**Why is this bad/What are the consequences**:
The fastest wizard round in the west.

**Steps to reproduce the problem**:
Start server, set race to Plasmaman, set wizard preference, force wizard round, burn and choke to death.

**When did the problem start happening**:
Probably when Plamaman got their rework. #11987

### LINKED ISSUE #13451: Refunding a spell issue when using admin tools
**Issue Description**:

If you refund a spell using a spellbook different than the spellbook you learned it from, the action button will not go away, while the spell will be properly removed/refunded. This can only happen with admin stuff currently since wiz can only refund on z2 anyway, but I'm documenting this as I gotta fix it for wizard loadouts PR to work properly.

**What did you expect to happen**: 

Refund spell, spell button be gone.

**What happened instead**:

Refund spell, spell button not gone

**Why is this bad/What are the consequences**:

Cursed useless button not going away.

**Steps to reproduce the problem**:

1. Use a one-use spellbook or an admin spawned spellbook to learn a spell (or wizard loadouts from my PR)
2. Refund using a different spellbook
3. Action button not removed

**When did the problem start happening**:

Probably an oversight that never came up when adding spell action buttons

**Extra information**:

Only tested on local, can only be reproduced using admin tools currently.

### PR DESCRIPTION
**Pull request by datlo**
Fixes #12668
Fixes #13451 

Thanks to Ionward for providing the new sprites in this PR

Adds a new tab to the spellbook : The loadout tab. It comes with two categories.

- Standard loadouts are suggestions aimed at new players. 
They provide 10 spell points worth of spells, at a cost of 10 points. These spells can be refunded.  The goal is to give players without wizard knowledge/experience some solid loadouts to work with, and hopefully reduce the amount of 5 minutes wizard rounds.

image

- Unique loadouts are gimmicky, 10 points loadouts that will destroy the spellbook once bought, preventing refund. (There is a warning confirmation prompt). They provide an unusual wizard playstyle, sometimes with items and/or spells that cant be bought in the spellbook on their own.

image

Gunslinging reaper outfit : 
image

Fixes spell icons not updating to show that den-restricted spells are now castable when you first teleport to the station.

Also fixes plasmaman burning to death roundstart when spawning as a wizard by adding a new wizard envirosuit and helmet for plasmaman wizards. It was also added to the den's racks, for players who wish to turn into plasmamen with the magic mirror.

image

Additionally, this PR adds a Mime Wizard outfit to the wizard vendor, and some extra spare clothes/robes to the den's dresser closet. Comes with species sprites for grey, drask and vox.
There is also a 'Mute' option in the magic mirror now.

image

Removed the spell list paper in the den, as it was out of date and redundant since the spellbook describe the spells you can get well enough.

:cl:
add: Added a new category to the wizard spellbook, Loadouts.
add: Added the Offense Focus : Mutant, Defense Focus : Lich, and Utility Focus : Wands standard loadouts.
add: Added the Silencio and Gunslinging Reaper unique loadouts.
add: Added Mime Wizard robes, sprited by Ionward.
add: Added a new envirosuit for plasmaman wizards, sprited by Ionward
fix: Plasmaman wizards no longer burn to death roundstart.
fix: Vox wizards no longer start the round damaged by toxic oxygen.
fix: Fixed spell action buttons not updating properly when teleporting from the wizard den to the station.
add: Added some additional spare clothes to the wizard den.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/wizard/wizard.dm [hunk 1]
```diff
@@ -135,30 +135,39 @@
 	qdel(wizard_mob.r_store)
 	qdel(wizard_mob.l_store)
 
+	if(!isplasmaman(wizard_mob))
```
Comment (farie82) [inline code/game/gamemodes/wizard/wizard.dm]: More readable if you turn this around.
If plasmaman do special case
else do normal case

File: code/game/gamemodes/wizard/wizloadouts.dm [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		</i>Provides a .357 Revolver, 4 speedloaders of ammo, Ethereal Jaunt, Blink, Summon Item, No Clothes, and Bind Soul, with a unique outfit.<i>"
+	log_name = "GR"
+	items_path = list(/obj/item/gun/projectile/revolver, /obj/item/ammo_box/a357, /obj/item/ammo_box/a357, /obj/item/ammo_box/a357, /obj/item/ammo_box/a357, /obj/item/clothing/under/syndicate)
+	spells_path = list(/obj/effect/proc_holder/spell/targeted/ethereal_jaunt, /obj/effect/proc_holder/spell/targeted/turf_teleport/blink, \
+		/obj/effect/proc_holder/spell/targeted/summonitem, /obj/effect/proc_holder/spell/noclothes, /obj/effect/proc_holder/spell/targeted/lichdom/gunslinger)
+	category = "Unique"
+	destroy_spellbook = TRUE
+
+/obj/effect/proc_holder/spell/targeted/lichdom/gunslinger/equip_lich(mob/living/carbon/human/H)
+		H.equip_to_slot_or_del(new /obj/item/clothing/suit/storage/det_suit(H), slot_wear_suit)
```
Comment (farie82) [inline code/game/gamemodes/wizard/wizloadouts.dm]: DOUBLE TAB!!!!

**Comment by taukausanake** [comment]
FIRE MAGE FIRE MAGE FIRE MAGE FIRE MAGE

Maybe a minion loadout too? That sounds nice but I don't know all the spell costs so I'm just spitballing ideas.

_please fire mage_

**Comment by Ionward** [comment]
there are those psy-amp robes that count as magic robes, you could do a gimmicky loadout as some psycaster if you wanted more options in that category

**Comment by nok-ko** [comment]
Looks awesome.

**Comment by datlo** [comment]
Conflict fixed

File: code/game/gamemodes/wizard/spellbook.dm [hunk 3]
```diff
@@ -571,6 +573,48 @@
...
+
+/datum/spellbook_entry/loadout/Buy(mob/living/carbon/human/user, obj/item/spellbook/book)
+	if(destroy_spellbook)
+		var/response = alert(user, "The [src] loadout cannot be refunded once bought. Are you sure this is what you want?", "No refunds!", "No", "Yes")
+		if(response == "No")
+			return FALSE
+		to_chat(user, "<span class='notice'>[book] crumbles to ashes as you acquire its knowledge.</span>")
+		qdel(book)
+	else if(items_path.len)
+		var/response = alert(user, "The [src] loadout contains items that will not be refundable if bought. Are you sure this is what you want?", "No refunds!", "No", "Yes")
```
Comment (Fox-McCloud) [inline code/game/gamemodes/wizard/spellbook.dm:605]: no `The [src]`; this will come out as "The the thing".

**Comment by Fox-McCloud** [comment]
Just the one thing.

**Comment by McRamon** [comment]
A little idea if you dont mind, special loadout.
May call it warlock, or demon lord, or something like it.

2 bottles of blood
Demon heart to bloodcrawl yourself
Some fancy suit that fits theme
Fireball spell
Dark blessing

Its not final ofc, may be tweaked, but the idea is to have blood crawling wiz loadout

**Comment by datlo** [comment]
@Fox-McCloud let me test how it looks IG but i believe its actually correct as-is because the loadout names are capitalised. testing rn.

**Comment by datlo** [comment]
image
image

File: code/game/gamemodes/wizard/spellbook.dm (hunk 3 above)
Comment (datlo) [reply code/game/gamemodes/wizard/spellbook.dm:605]: Loadout names are capitalised so they dont get 'the' appended by code when using [src]

**Comment by datlo** [comment]
Found a bug, spell buttons arent removed when the spell is refunded, fixing.

**Comment by datlo** [comment]
Fixed. This is ready for merge

**Pull request merged**